### PR TITLE
Issue 660: CodeSnippet Reuseable component

### PR DESCRIPTION
**Pull request by NWCalvank**
## Hey, I just made a Pull Request!

This adds a very minimal wrapper around [react-syntax-highlighter](https://github.com/conorhastings/react-syntax-highlighter) to produce a reusable component that automatically observes light/dark mode.

This is my first contribution to Backstage, so please advise should my implementation not be fully-aligned with the vision for this component/project. Thanks!

## Features
- Toggleable line numbers
- Automatic light/dark mode support
- Basic test coverage for rendering without runtime errors & with/out line numbers
- Simple stories for potential configurations & common languages

## Notes
- There are [many props](https://github.com/conorhastings/react-syntax-highlighter#props) available that could be observed. With the exception of `showLineNumbers`, none have been touched, so all default values will be used.
- There are also [many languages](https://github.com/conorhastings/react-syntax-highlighter/blob/master/AVAILABLE_LANGUAGES_HLJS.MD) and [many styles](https://github.com/conorhastings/react-syntax-highlighter/blob/master/AVAILABLE_STYLES_HLJS.MD) supported by this component. I have chosen a couple of options that seemed safe and fell in line with someone else's thinking [here](https://github.com/spotify/backstage/pull/662/files#diff-d6eb8446a83ddb7ba0c8110bf5786d77R30)
- The (heavy) default syntax library is being used here. We may opt for the [light version](https://github.com/conorhastings/react-syntax-highlighter#light-build) if we are concerned about the size of this library
- The default syntax library is used in order to honour light mode. Another option [Prism](https://github.com/conorhastings/react-syntax-highlighter#prism) also exists, which provides JSX support but does not provide a light mode at this time
- The commonJS build is being used here as Jest does not appear to be configured with ESM support in Backstage at this time
- As per this [PR comment](https://github.com/spotify/backstage/pull/662/files#r416553092), no, there does not appear to be support for importing the syntax libraries in a more typical style

## Areas for Improvement
- There is no automated test coverage for the light/dark mode support, as testing the style attributes directly felt brittle. I did not think of a more reasonable solution in the absence of snapshot tests. If there is already snapshot test support, I was unable to find it.

Simple Multiline - Light
Screen Shot 2020-05-10 at 2 53 21 PM

Simple Multiline - Dark
Screen Shot 2020-05-10 at 2 53 08 PM

Language Examples - Light
Screen Shot 2020-05-10 at 2 54 01 PM

Language Examples - Dark
Screen Shot 2020-05-10 at 2 53 48 PM

#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [x] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [x] Prettier run on changed files
- [x] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: packages/core/src/components/CodeSnippet/CodeSnippet.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    >
+      {text}
+    </SyntaxHighlighter>
+  );
+};
+
+// Type check for the JS files using this core component
+CodeSnippet.propTypes = {
+  text: PropTypes.string.isRequired,
+  language: PropTypes.string.isRequired,
```
Comment (freben) [inline packages/core/src/components/CodeSnippet/CodeSnippet.tsx:57]: Add `showLineNumbers` here too (but not required).

**Review by freben — COMMENTED** [review]
This was really well made, thank you. Just a nitpick and then I'm good.

**Review by stefanalund — APPROVED** [review]
Thanks a lot for this excellent contribution! Welcome to the Backstage community 🎉

**Comment by stefanalund** [comment]
Closes https://github.com/spotify/backstage/issues/660

**Pull request merged**